### PR TITLE
OHRM5X-1964: Initiate oauth - authorization code flow

### PR DESCRIPTION
**Pull request by ManulMax**

### CODE REVIEW

File: src/layouts/DrawerContent.tsx [hunk 1]
```diff
@@ -132,8 +132,8 @@ const DrawerContent = (props: DrawerContentProps & DrawerItemListProps) => {
   return (
     <SafeAreaView style={styles.safeArea}>
       <ProfilePicture
-        name={myInfo?.employee.fullName}
-        jobTitle={myInfo?.employee.jobTitle}
+        name={myInfo?.employee.firstName + ' ' + myInfo?.employee.lastName}
```
Comment (RajithaKumara) [inline src/layouts/DrawerContent.tsx]: will create helper method

File: src/screens/login/Login.tsx
Comment (RajithaKumara) [inline src/screens/login/Login.tsx:1]: will remove this file, and route related to this and types

File: src/services/endpoints.ts [hunk 2]
```diff
@@ -53,12 +53,20 @@ export const API_ENDPOINT_ATTENDANCE_LIST =
 export const API_ENDPOINT_ATTENDANCE_CONFIGURATION = '/api/v1/time/config';
 
 //help
-export const API_ENDPOINT_HELP_CONFIG = '/api/v1/help/config';
+// export const API_ENDPOINT_HELP_CONFIG = '/api/v1/help/config';
```
Comment (RajithaKumara) [inline src/services/endpoints.ts]: will remove API call related

File: src/store/auth/reducer.ts [hunk 3]
```diff
@@ -67,6 +68,7 @@ const authReducer = (
       return {
         ...state,
         checkingInstance: false,
+        isAuthenticated: true,
```
Comment (RajithaKumara) [inline src/store/auth/reducer.ts]: need to check and will separate to new action (under store/storage/reducer ; username)

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: src/store/saga-effects/api.ts [hunk 4]
```diff
@@ -169,3 +169,58 @@ export function* apiPostCall(endpoint: string, body: object) {
...
+    const data = yield call([response, response.json]);
+    data.getResponse = () => {
+      return response;
+    };
+    return data;
+  }
+  throw new Error("Couldn't call with empty instanceUrl or accessToken.");
+}
+
+export function* apiDeleteCall(endpoint: string) {
```
Comment (RajithaKumara) [inline src/store/saga-effects/api.ts]: missing delete body

File: src/store/help/sagas.ts
Comment (RajithaKumara) [inline src/store/help/sagas.ts:1]: remove this

File: src/layouts/DrawerContent.tsx [hunk 5]
```diff
@@ -103,14 +94,7 @@ const DrawerContent = (props: DrawerContentProps & DrawerItemListProps) => {
   }
 
   const onPressHelp = () => {
```
Comment (RajithaKumara) [inline src/layouts/DrawerContent.tsx:97]:
```jsx
  const onPressHelp = () => {
    Linking.canOpenURL(HELP_REDIRECT_URL).then((supported) => {
      if (supported) {
          Linking.openURL(HELP_REDIRECT_URL);
      }
    });
  };
```

File: src/layouts/DrawerContent.tsx [hunk 6]
```diff
@@ -121,19 +105,12 @@ const DrawerContent = (props: DrawerContentProps & DrawerItemListProps) => {
...
-  /* eslint-enable react-hooks/exhaustive-deps */
 
   return (
     <SafeAreaView style={styles.safeArea}>
       <ProfilePicture
-        name={myInfo?.employee.fullName}
-        jobTitle={myInfo?.employee.jobTitle}
+        name={myInfo?.employee.firstName + ' ' + myInfo?.employee.lastName}
+        jobTitle={myInfo?.employee.jobTitle.title}
         imageSource={imageSource}
```
Comment (RajithaKumara) [inline src/layouts/DrawerContent.tsx:117]: TODO

File: src/store/auth/sagas.ts [hunk 7]
```diff
@@ -157,16 +165,14 @@ function* checkInstance(action?: CheckInstanceAction) {
...
 
       checkInstanceCompatibility(data);
-      checkRemovedEndpoints(data);
-      const usingDeprecatedEndpoints = checkDeprecatedEndpoints(data);
-      if (usingDeprecatedEndpoints) {
-        yield showSnackMessage('Please Update the Application.', TYPE_WARN);
-      }
 
-      yield* fetchEnabledModules();
+      // TODO
```
Comment (RajithaKumara) [inline src/store/auth/sagas.ts:172]: TODO

File: src/store/auth/sagas.ts [hunk 8]
```diff
@@ -237,16 +243,34 @@ function* fetchEnabledModules(action?: FetchEnabledModulesAction) {
...
+      const enabledModules: EnabledModules = {
+        modules: responseData.data,
+        meta: {
+          leave: {
+            isLeavePeriodDefined: true, // TODO
+          },
+          time: {
+            isTimesheetPeriodDefined: true, // TODO
+          },
+        },
```
Comment (RajithaKumara) [inline src/store/auth/sagas.ts:267]: TODO

File: src/store/auth/sagas.ts [hunk 9]
```diff
@@ -237,16 +243,34 @@ function* fetchEnabledModules(action?: FetchEnabledModulesAction) {
...
+      };
 
       if (responseData.data) {
-        yield put(fetchEnabledModulesFinished(responseData.data));
-        if (!responseData.data.modules.mobile) {
+        yield put(fetchEnabledModulesFinished(enabledModules));
+        if (!enabledModules.modules.mobile) {
```
Comment (RajithaKumara) [inline src/store/auth/sagas.ts:272]: TODO

File: src/store/auth/sagas.ts [hunk 10]
```diff
@@ -278,40 +302,38 @@ function* fetchEnabledModules(action?: FetchEnabledModulesAction) {
...
+          tokenEndpoint: authParams.instanceUrl + '/oauth2/token',
+        },
+        clientId: PUBLIC_MOBILE_CLIENT_ID,
+        redirectUrl: OAUTH_CALLBACK_URL,
+        additionalParameters: {},
+        usePKCE: true,
+        useNonce: false,
+        additionalHeaders: {Accept: 'application/json'},
+        connectionTimeoutSeconds: 5,
+        iosPrefersEphemeralSession: true,
```
Comment (RajithaKumara) [inline src/store/auth/sagas.ts:324]: TODO:: invalid in types

File: src/store/auth/sagas.ts [hunk 11]
```diff
@@ -350,15 +372,24 @@ function* fetchMyInfo() {
...
-          yield put(fetchMyInfoFinished(response.data));
+        if (response.data) {
+          const data: MyInfo = {
+            employee: response.data,
+            employeePhoto: null,
+            user: {
+              // TODO::remove
+              userRole: 'Admin',
+              isSupervisor: false,
+            },
```
Comment (RajithaKumara) [inline src/store/auth/sagas.ts:390]: TODO

File: src/store/auth/sagas.ts [hunk 12]
```diff
@@ -395,38 +426,26 @@ function* fetchMyInfo() {
   }
 }
 
-function* fetchApiDefinition(
-  action: FetchMyInfoAction | FetchNewTokenFinishedAction,
-) {
+function* fetchApiDefinition() {
```
Comment (RajithaKumara) [inline src/store/auth/sagas.ts:429]:
```suggestion
function* fetchRestApiVersion() {
```

File: src/store/auth/sagas.ts [hunk 13]
```diff
@@ -395,38 +426,26 @@ function* fetchMyInfo() {
...
-    if (action.type === FETCH_MY_INFO) {
-      const apiDetails: ApiDetails = yield select(selectApiDetails);
-      if (
-        apiDetails[INSTANCE_API_VERSION] !== null ||
-        apiDetails[INSTANCE_API_VERSION] !== undefined
-      ) {
-        return;
-      }
+    const apiDetails: ApiDetails = yield select(selectApiDetails);
+    if (
```
Comment (RajithaKumara) [inline src/store/auth/sagas.ts:433]: TODO

File: src/store/auth/types.ts [hunk 14]
```diff
@@ -136,9 +128,9 @@ export interface Employee {
   middleName: string;
   fullName: string;
   employeeId: string;
-  code: string;
-  jobTitle: NullableString;
-  unit: NullableString;
+  code: string | null;
+  jobTitle: JobTitle;
+  unit: SubUnit;
   supervisor: null | Supervisor[];
```
Comment (RajithaKumara) [inline src/store/auth/types.ts:134]: TODO:: refactor remove unnecessary

File: src/store/auth/types.ts [hunk 15]
```diff
@@ -147,16 +139,8 @@ export interface EmployeePhoto {
 }
 
 export interface User {
-  userName: string;
   userRole: typeof USER_ROLE_ADMIN | typeof USER_ROLE_ESS;
   isSupervisor: boolean;
```
Comment (RajithaKumara) [inline src/store/auth/types.ts:143]: TODO:: find alternative way to handle this

**Review by RajithaKumara — APPROVED** [review]

File: src/screens/login/Login.tsx
Comment (ManulMax) [reply src/screens/login/Login.tsx:1]: removed

**Pull request merged**